Pomodoro.Services/Realizations/SettingsService.cs

```diff
@@ -0,0 +1,90 @@
+﻿using AutoMapper;
+using Pomodoro.Core.Interfaces.IServices;
+using Pomodoro.Core.Models;
+using Pomodoro.DataAccess.Entities;
+using Pomodoro.DataAccess.Repositories.Interfaces;
+
+namespace Pomodoro.Services.Realizations
+{
+    public class SettingsService : ISettingsService
+    {
+        private readonly IMapper mapper;
+        private readonly ISettingsRepository settingsRepository;
+
+        public SettingsService(
+            IMapper mapper,
+            ISettingsRepository settingsRepository)
+        {
+            this.mapper = mapper;
+            this.settingsRepository = settingsRepository;
+        }
+
+        public async Task<SettingsModel> CreateSettingsAsync(SettingsModel settingsModel)
+        {
+            if (settingsModel is null)
+            {
+                throw new ArgumentNullException(nameof(settingsModel));
+            }
+
+            var settings = mapper.Map<Settings>(settingsModel);
+
+            await RemoveExistingSettings(settings.UserId);
+            await settingsRepository.AddAsync(settings);
+            await settingsRepository.SaveChangesAsync();
+
+            return mapper.Map<SettingsModel>(settings);
```

Comment:
Is mapper.Map<SettingsModel>(settings) value not the same settingsModel?

Reply:
They are the same, except for the Id which can be empty initially. But after adding to the db it will be generated. Maybe such an approach would be clearer:

`settingsModel.Id = settings.Id;`
`return settingsModel;`

But it has a downside. Because if new auto-generated properties are added to this entity in the future, we'll have to add a few more same lines of code to this method. That doesn't seem right to me, so I'd like to leave it as it is.